plugins/BEdita/Core/src/Model/Entity/Stream.php

```diff
@@ -289,9 +296,24 @@ protected function readFileMetadata($resource): void
         }
 
         rewind($resource);
-        $exif = exif_read_data($resource);
-        if (!empty($exif)) {
-            $this->file_metadata = $exif;
+        $exif = exif_read_data($resource, null, true);
+        if ($exif === false) {
+            return;
         }
+
+        // Filter non-standard sections
+        $exif = array_intersect_key($exif, array_flip(static::EXIF_SECTIONS));
+        // Filter undefined tags, which may break JSON encoding
+        $exif = array_map(function ($section) {
+            return array_filter($section, function ($key) {
+                return strpos($key, 'UndefinedTag:') === false;
+            }, ARRAY_FILTER_USE_KEY);
+        }, $exif);
+        // Check if data is JSON-encodable
+        if (json_decode(json_encode($exif)) === null) {
```

Comment:
To check if `$exif` can be successfully JSON-encoded, it should be sufficient to check if `json_encode()` returns `false`:

```suggestion
        if (json_encode($exif) === false) {
```